packages/cli/src/ui/hooks/useModelCommand.ts

```diff
@@ -0,0 +1,42 @@
+/**
+ * @license
+ * Copyright 2025 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { useState, useCallback } from 'react';
+import type { Config } from '@google/gemini-cli-core';
+
+interface UseModelCommandReturn {
+  isModelDialogOpen: boolean;
+  openModelDialog: () => void;
+  closeModelDialog: () => void;
+  handleModelSelect: (model: string) => void;
+}
+
+export const useModelCommand = (config: Config): UseModelCommandReturn => {
+  const [isModelDialogOpen, setIsModelDialogOpen] = useState(false);
+
+  const openModelDialog = useCallback(() => {
+    setIsModelDialogOpen(true);
+  }, []);
+
+  const closeModelDialog = useCallback(() => {
+    setIsModelDialogOpen(false);
+  }, []);
+
+  const handleModelSelect = useCallback(
```

Comment:
any reason to not just move this directly within modelDialog? it already knows about the config. Might as well not separate stuff that can just be in one place. will simplify appContainer rather than tangling it with impl details. as a follow up we should regularize all the openDialog, closeDialog spew in AppContainer with generic functionality for that for all dialogs rather than creating so much boilerplate.